File: Structure-syntax/15 Procedure/C++/src/main.cpp
/*
 * C++ program
 */
#include <iostream>
using namespace std;

void Proc (int Var1, int & Var2, int & Var3);

int Func (int Var);

void Proc (int Var1, int & Var2, int & Var3) {
    Var2 = Func (Var1);
    Var3 = Var3 + 1;
}

int Func (int Var) {
    return Var + 1;
}

int main (int argc, const char *argv[]) {
   ;  // эквивалент null оператора Ada
}

File: Structure-syntax/15 Procedure/Java/Main.java
/*
* Java program
*/
public class Main {
   public static void main(String [] argv) {

      class ProcData {
         public int Var2;
         public int Var3;

         public void Proc (int Var1) {
            Var2 = Func (Var1);
            Var3 = Var3 + 1;
         }

         int Func (int Var) {
            return Var + 1;
         }
      }
   }
}
